### ISSUE TITLE
 Incompatible with eslint ts config

### ISSUE THREAD

**Issue by BTMuli** [ota-meshi/eslint-plugin-jsonc#385]
Thanks for your plugin,it helps me lint json file easily.😍

Since Eslint now supports [typescript config files](https://eslint.org/docs/latest/use/configure/configuration-files#typescript-configuration-files),I found type error while using this plugin.

![image](https://github.com/user-attachments/assets/9bb4eac0-6b6d-48b2-89fd-7a71c57663e1)

**Comment by high1** [comment]
I'm getting this error with `"exactOptionalPropertyTypes": true` files in my typescript configuration. Can you confirm that's the case, @BTMuli?

**Comment by csantos-nydig** [comment]
I believe I'm getting the same error

<img width="1366" alt="Image" src="https://github.com/user-attachments/assets/3fcedc09-f1fa-441d-8958-bc2d2efc41af" />

**Comment by barrymichaeldoyle** [comment]
> I'm getting this error with `"exactOptionalPropertyTypes": true` files in my typescript configuration. Can you confirm that's your case also, [@BTMuli](https://github.com/BTMuli)?

Yes, this becomes an issue when you set that `exactOptionalPropertyTypes` to `true`. We'd love it if this was resolved. Currently we work around it by typecasting it `as unknown as Linter.Config[]`, as opposed to an ugly series of spreads.
